pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -1096,8 +1100,43 @@ func KeyV2(
 		data = append(data, "node-local-dns")
 	}
 
-	if !inPlaceUpdate {
-		data = append(data, gardenerutils.CalculateDataStringForKubeletConfiguration(kubeletConfiguration)...)
+	data = append(data, gardenerutils.CalculateDataStringForKubeletConfiguration(kubeletConfiguration)...)
+
+	var result string
+	for _, v := range data {
+		result += utils.ComputeSHA256Hex([]byte(v))
+	}
+
+	return fmt.Sprintf("gardener-node-agent-%s-%s", worker.Name, utils.ComputeSHA256Hex([]byte(result))[:16])
+}
+
+// KeyForInPlaceUpdates returns the key that can be used as secret name for worker with in-place updates.
+//
+// WARNING: Modifying fields in this function will change the node agent secret name,
+// causing the running GNA to lose access to its secret. This will result in the GNA entering a broken state.
+// Therefore, changes to these fields should be denied during validation.
+func KeyForInPlaceUpdates(worker *gardencorev1beta1.Worker, nodeLocalDNSEnabled bool) string {
```

Comment:
What is the approach for future adaptations? I understand the key may never change for existing shoots with in-place update pools, but we might still introduce new fields. How would we handle this?

Generally, if the key must remain static anyways, why do we perform such a complicated computation? Can't we just hard-code it?

Reply:
> Generally, if the key must remain static anyways, why do we perform such a complicated computation? Can't we just hard-code it?

Yes, there is no plan to change the hash at all. We kept only those fields here in function, changing which would be impossible, so we can keep track them.
Other option would be to just keep the worker name only here for hash calculation.

Reply:
If the name can never change, why do you propose this "complex" computation here in the first place? What is the benefit/motivation of that?

Reply:
Just to keep track, what are the fields, which will always be static no matter what in future.